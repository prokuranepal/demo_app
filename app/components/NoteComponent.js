import React from "react";
import { Text, StyleSheet, View } from "react-native";

import PropTypes from 'prop-types';

import {fontSize} from '../theme/fonts';
import {colors} from '../theme/themeColors';

const NoteComponent = props => (
    <View  style={{...styles.noteView, borderLeftColor:props.color}} data-test="noteComp" >
        <Text style={[styles.textHeading, props.style]} data-test="headingComp">
             {props.title}
         </Text>
        <Text style={[styles.textNormal, props.style]} data-test="descComp">
             {props.children}
         </Text>
  </View>
);

const styles = StyleSheet.create({
    noteView:{
    padding:10,
    paddingTop:12,
    marginTop:10,
    marginBottom:20,
    borderLeftColor: colors.defaultBorder,
    borderLeftWidth:10  ,
    backgroundColor: colors.note
    },

  textNormal: {
    fontSize:fontSize.noteDescription, 
    color:colors.textColor1, 
  },
  
  textHeading: {
    fontSize:fontSize.noteHeading, 
    color: colors.textColor1, 
    fontWeight:"bold"
  }
});

NoteComponent.propTypes = {
  style: PropTypes.object,
  children: PropTypes.any,
  color: PropTypes.string,
  title: PropTypes.string
}

export default NoteComponent;