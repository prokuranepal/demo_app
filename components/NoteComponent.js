import React from "react";
import { Text, StyleSheet, View } from "react-native";

const noteComponent = props => (
    <View  style={{...styles.noteView, borderLeftColor:props.color}} >
        <Text style={[styles.textHeading, props.style]} >
             {props.title}
         </Text>
        <Text style={[styles.textNormal, props.style]} >
             {props.children}
         </Text>
  </View>
);

const styles = StyleSheet.create({
    noteView:{
    padding:10,
    marginVertical:20,
    borderLeftColor: "#464f7a",
    borderLeftWidth:10  ,
    backgroundColor:"#ebebec" 
    },

  textNormal: {
    fontSize:15, 
    color:"#575758", 
  },
  
  textHeading: {
    fontSize:18, 
    color:"#575758", 
    fontWeight:"bold"
  }
});

export default noteComponent;