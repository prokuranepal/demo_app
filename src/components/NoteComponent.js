import React from "react";
import { Text, StyleSheet, View } from "react-native";
import {colors} from '../ThemeColors/themeColors';
const noteComponent = props => (
    <View  style={{...styles.noteView, borderLeftColor:props.color}} data-test="noteComp">
        <Text style={[styles.textHeading, props.style]} data-test="headingComp">
             {props.title}
         </Text>
        <Text style={[styles.textNormal, props.style]} data-test="descComp">
             {props.children}
         </Text>
  </View>
);

const styles = StyleSheet.create({
    noteView:{
    padding:10,
    paddingTop:12,
    marginTop:10,
    marginBottom:20,
    borderLeftColor: colors.defaultBorder,
    borderLeftWidth:10  ,
    backgroundColor: colors.note
    },

  textNormal: {
    fontSize:15, 
    color:colors.textColor1, 
  },
  
  textHeading: {
    fontSize:18, 
    color: colors.textColor1, 
    fontWeight:"bold"
  }
});

export default noteComponent;