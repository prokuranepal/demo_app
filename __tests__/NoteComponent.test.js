import React from 'react';
import {View, Text} from 'react-native';
import {
    configure,
    shallow
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import NoteComponent from '../app/components/NoteComponent';
import {findByTestAttr} from '../app/utils/testFunction';
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})

const dummy_data = {
    title:"Note Header",
}


describe('<NoteComponent/>', () => {

    it('should render without issues', () => {
        const component = shallow(<NoteComponent />);
        expect(component.length).toBe(1);
        expect(component).toMatchSnapshot();
      });

    it("Header Props and children test in NoteComponent", () =>
      {
          const wrapper = shallow(<NoteComponent {...dummy_data}>Headerrr Description</NoteComponent>)
          expect(wrapper.find(View)).toHaveLength(1)          
          expect(wrapper.find(Text)).toHaveLength(2)
          let headingComp = findByTestAttr(wrapper, "headingComp")
          expect(headingComp.length).toBe(1);

          expect(headingComp.props().children).toEqual("Note Header")
          let descComp  = findByTestAttr(wrapper, "descComp")
          expect(descComp.length).toBe(1);
          expect(descComp.props().children).toEqual("Headerrr Description")

      
     });

})