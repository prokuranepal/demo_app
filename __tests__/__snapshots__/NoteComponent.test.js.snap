// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoteComponent/> should render without issues 1`] = `ShallowWrapper {}`;
